server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -727,7 +727,13 @@ private void setMaxDocvalueFields(int maxDocvalueFields) {
      */
     public int getHighlightMaxAnalyzedOffset() { return this.maxAnalyzedOffset; }
 
-    private void setHighlightMaxAnalyzedOffset(int maxAnalyzedOffset) { this.maxAnalyzedOffset = maxAnalyzedOffset; }
+    private void setHighlightMaxAnalyzedOffset(int maxAnalyzedOffset) {
+        if (maxAnalyzedOffset < 1) {
+            throw new IllegalArgumentException(
+                "[" + MAX_ANALYZED_OFFSET_SETTING.getKey() + "] must be >= 1");
+        }
+        this.maxAnalyzedOffset = maxAnalyzedOffset;
```

Comment:
@jimczi  I have added a custom validator here, as you suggested. This should guard against setting value <1.  Can you please continue the reviewing when you have available time.